roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,453 @@
+    of the statements backing it.
+-   Can be one of two kinds:
+    -   `Full`
+        -   Contains information about the candidate and should be sent to peers who
+            may not have the candidate yet.
+    -   `Acknowledgement`
+        -   Omit information implicit in the candidate, and should be sent to peers
+            which are guaranteed to have the candidate already.
+
+### Manifest exchange
+
+-   Occurs between us and a peer in the grid.
+-   Why:
+    -   Indicates that both nodes know the candidate as valid and backed.
+    -   Allows the nodes to send `Statement` messages directly to each other for any
+        new statements they might need.
+-   Both `manifest_sent_to` and `manifest_received_from` have been invoked.
+-   In practice, it means that one of three things have happened:
+    -   They announced, we acknowledged
+    -   We announced, they acknowledged
+    -   We announced, they announced (not sure if this can actually happen; it would
+        happen if 2 nodes had each other in their sending set and they sent
+        manifests at the same time. The code accounts for this anyway)
+-   After conclusion, we update pending statements.
+    -   We now know our statements and theirs.
+    -   Pending statements are those we know locally that the remote node does not.
+
+## Cluster
+
+-   Direct distribution of unbacked candidates within a group.
+-   Why:
+    -   because of prospective parachains communication has to happen between
+        backing groups on rotation boundaries.
+    -   Only have to send unbacked candidates within groups.
+-   Bound number of `Seconded` messages per validator per relay-parent.
+    -   Why: spam prevention
+    -   Validators can try to circumvent this, but they would only consume a few KB
+        of memory and it is trivially slashable on chain.
+-   `ClusterTracker`
+    -   Determine whether to accept/reject messages from other validators in the
+        same group
+    -   Keep track of what we have sent to other validators in the group and pending
+        statements.
+-   Protocol: See "Protocol"
+
+## Grid
+
+-   Distribution of backed candidates and late statements outside the group.
+-   Protocol: See "Protocol"
+
+### Grid topology
+
+-   The gossip topology
+    -   Why: limit the amount of peers we send messages to and handle view updates.
+-   The basic operation of the 2D grid topology is that:
+    -   A validator producing a message sends it to its row-neighbors and its
+        column-neighbors
+    -   A validator receiving a message originating from one of its row-neighbors
+        sends it to its column-neighbors
+    -   A validator receiving a message originating from one of its column-neighbors
+        sends it to its row-neighbors
+-   This grid approach defines 2 unique paths for every validator to reach every
+    other validator in at most 2 hops.
+-   However, we also supplement this with some degree of random propagation.
```

Comment:
Actually, we don't, not in this case, where backing candidates is not critical. We do do this in approval-distribution, where it's of the utmost importance that all messages arrive. Only downside to stuff getting dropped here is fewer candidates getting backed. annoying, but not the end of the chain.

Reply:
Thanks for catching that. I actually didn't see anything about random propagation in `statement-distribution` itself, just in `gossip-support`.